Car database

mykolamuntian opened the pull request:
Task:
- Implement TODO from a class example
- Test MemoryDatabase in main
- *Perform serialization to a file and deserialization of the car database using the mechanisms ObjectInputStream and ObjectOutputStream

mykolamuntian committed:
Added main classes

mykolamuntian committed:
Added more fields to describe Driver
Redesigned equals() and hashCode() methods

mykolamuntian committed:
Work on method findAll()

mykolamuntian committed:
Finished findAll() using for each

mykolamuntian committed:
Added test methods
Added method printDataBase() to show changes in data base
Added one more value of OperationResult
Override toString of Car and CarNumber to print them

strazhkoe commented:
Опишите пожалуйста, какой код сделан Вами. Сливается Ваш и мой код и непонятно что смотреть

mykolamuntian committed:
Added markers near my code

mykolamuntian committed:
Added method clone() in classes-entities
Added test of this methods
Added distance to the method toString in the class Car

mykolamuntian committed:
Implemented interface Serializable in MemoryDataBase and in all entities
Added method of serialization of car data base in Main
